src/reducers/paths.reducer.js

```diff
@@ -18,26 +18,51 @@ import {
   IMPORT_EXISTING_PROJECT_FINISH,
   FINISH_DELETING_PROJECT,
   RESET_ALL_STATE,
+  CHANGE_PROJECT_HOME_PATH,
 } from '../actions';
 import { windowsHomeDir, isWin } from '../services/platform.service';
 
 import type { Action } from 'redux';
 
 type State = {
-  [projectId: string]: string,
+  homePath: string,
+  byId: {
+    [projectId: string]: string,
+  },
 };
 
-const initialState = {};
+const homedir = isWin ? windowsHomeDir : os.homedir();
+// Noticing some weird quirks when I try to use a dev project on the compiled
+// "production" app, so separating their home paths should help.
+
+const initialState = {
+  homePath:
+    process.env.NODE_ENV === 'development'
+      ? path.join(homedir, 'guppy-projects-dev')
+      : path.join(homedir, 'guppy-projects'),
+  byId: {},
+};
 
 export default (state: State = initialState, action: Action) => {
   switch (action.type) {
     case ADD_PROJECT:
     case IMPORT_EXISTING_PROJECT_FINISH: {
       const { projectPath, project } = action;
+      return {
+        ...state,
+        byId: {
+          ...state.byId,
+          [project.guppy.id]:
+            projectPath || `${getProjectHomePath(state)}/${project.guppy.id}`,
+        },
+      };
```

Comment:
Also, I have a few issues with the second half of this conditional:

- we should use `path.join` to be Windows-compatible (it uses `\` instead of `/` as the delineater)
- by convention, `getX` methods are usually selectors that receive the entire Redux state as a value, and not just a reducer's slice of it
- it feels like a pretty "busy" expression, to have a function call within a string interpolation within a boolean conditional within a setter.

What do you think of this alternative?

```js
const projectPath = action.projectPath || formatProjectPath(state.homePath, project.guppy.id);
state.byId[project.guppy.id] = projectPath;

// Further down, under `helpers`:
const formatProjectPath = (homePath, projectId) => path.join(homePath, projectId)
```

This way, the reducer code is nice and simple, and we're hiding the path business in a helper method (that can't be confused with a selector)

Reply:
@joshwcomeau there is a small typo in the `Immer` code - I think it should be `draftState.byId[project.guppy.id]` instead of `state`.

Reply:
OK, I will resolve them within today,  thank for your review @joshwcomeau @AWolf81 .